### PR TITLE
Function tag

### PR DESCRIPTION
**Pull request by tomoki0120**
# WHAT

Add_tag_function
# WHY

You can tag when you express your emotion for proto.
# DEADLINE

10/19

### CODE REVIEW

**Comment by tomoki0120** [comment]
https://gyazo.com/a78a77134b470f388d749306b9c4342c

**Comment by tomoki0120** [comment]
https://gyazo.com/9f656a1f6f9912ddb59bb5fdd2d1384f

File: app/controllers/prototypes_controller.rb [hunk 1]
```diff
@@ -45,7 +45,7 @@ def update
 
   private
   def proto_params
-    params.require(:proto).permit(:title, :catch_copy, :concept, images_attributes: [:id, :photo, :status]).merge(user_id: current_user.id)
+    params.require(:proto).permit(:title, :catch_copy, :concept, images_attributes: [:id, :photo, :status]).merge(user_id: current_user.id,tag_list: params[:proto][:tag])
   end
```
Comment (takuma3110) [inline app/controllers/prototypes_controller.rb:49]: good!

File: app/controllers/tags_controller.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class TagsController < ApplicationController
+
+  def index
+    @tags = ActsAsTaggableOn::Tag.most_used
```
Comment (takuma3110) [inline app/controllers/tags_controller.rb]: tagのindexはtagの一覧が出るようにしよう！

Comment (takuma3110) [reply app/controllers/tags_controller.rb]: https://gyazo.com/c21b0affb570b49655ef130a9205a883

File: app/controllers/tags_controller.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  end
+
+  def show
+    @tag = ActsAsTaggableOn::Tag.where(name: tag_params)
```
Comment (takuma3110) [inline app/controllers/tags_controller.rb]: findでいけるね

File: app/controllers/tags_controller.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    @prototypes = Proto.tagged_with(@tag).pagination_prototype_per(params[:page])
+  end
+
+  private
+  def tag_params
```
Comment (takuma3110) [inline app/controllers/tags_controller.rb]: findでやればtag_paramsは定義しなくてもいいね

**Comment by tomoki0120** [comment]
https://gyazo.com/38342ec2aac1a76e39c07bda75fda472

**Comment by tomoki0120** [comment]
indexをtomato-spaceのhtml/cssぽくしました！

**Comment by takuma3110** [comment]
image

**Pull request merged**